components/feature/prompts/src/main/java/mozilla/components/feature/prompts/concept/SelectablePromptView.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.concept
+
+import android.view.View
+
+/**
+ * An interface for views that can display an option selection prompt.
+ */
+interface SelectablePromptView<T> {
+
+    var listener: Listener<T>?
+
+    /**
+     * Shows an option selection prompt with the provided options.
+     *
+     * @param options A list of options to display in the prompt.
+     */
+    fun showPrompt(options: List<T>)
+
+    /**
+     * Hides the option selection prompt.
+     */
+    fun hidePrompt()
+
+    /**
+     * Casts this [SelectablePromptView] interface to an Android [View] object.
+     */
+    fun asView(): View = (this as View)
+
+    /**
+     * Tries to inflate the view if needed.
+     *
+     * See: https://github.com/mozilla-mobile/android-components/issues/5491
+     *
+     * @return true if the inflation was completed, false if the view was already inflated.
+     */
+    fun tryInflate(): Boolean
```

Comment:
We shouldn't need this any more, so let's remove it from the API. It was a technical detail that (in general) we don't need to expose.

In this case, we can even remove that code too.

Reply:
To be addressed in my immediate work on https://github.com/mozilla-mobile/android-components/issues/9457, so that this removal can be reviewed.